public/API/API_GetGameFromHash.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ *  API_GetGameFromHash - looks up a game from a given hash string
+ *    h : game hash
+ *
+ *  int        ID                         id of the game
+ *  string     Title                      name of the game
+ *  int        ConsoleID                  unique identifier of the console associated to the game
+ *  string     ConsoleName                name of the console associated to the game
+ */
+
+use App\Platform\Models\GameHash;
+use App\Platform\Models\System;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Validator;
+
+$input = Validator::validate(Arr::wrap(request()->query()), [
+    'h' => ['required', 'size:32'],
+]);
+
+$givenHash = request()->query('h');
+
+$foundHash = GameHash::with('game')->where('MD5', $givenHash)->first();
+
+if (!$foundHash) {
+    return response()->json(['message' => 'Could not find a game matching the given hash.'], 400);
```

Comment:
not found should return 404.

Reply:
Doesn't this imply the endpoint itself doesn't exist? In this case, the request is successful, but the server can't do anything with what the client has provided (4xx).

Reply:
Updated in https://github.com/RetroAchievements/RAWeb/pull/2090/commits/8ec90f1122c9f8f7847bdfdebacc32a8be1e9d4a.

Reply:
fwiw, hitting a non-existing endpoint returns the error in a nested structure.
```
{"message":"Not Found","errors":[{"status":404,"code":"not_found","title":"Not Found"}]}
```

https://github.com/RetroAchievements/RAWeb/blob/cf9953f7181df2c45a44ced1d39a10f30b3d5f2f/app/Exceptions/Handler.php#L101-L110

Reply:
Adjusted in latest to follow a similar structure:

```json
{
	"message": "Unknown hash: 2bc674be034e43c96b86487ac69d9293",
	"errors": [
		{
			"status": 404,
			"code": "not_found",
			"title": "Unknown hash: 2bc674be034e43c96b86487ac69d9293"
		}
	]
}
```